tests/test_aea_builder.py

```diff
@@ -1018,3 +1018,98 @@ def test_builder_pypi_dependencies():
         "open-aea-ledger-ethereum",
         "open-aea-ledger-cosmos",
     }
+
+
+class TestApplyEnvironmentVariables(AEATestCase):
+    """Test builder set from project dir, to make a skill 'abstract'."""
+
+    path_to_aea = Path(CUR_PATH) / "data" / "dummy_aea"
+    override = dedent(
+        """
+        ---
+        public_id: dummy_author/dummy:0.1.0
+        type: skill
+        models:
+          dummy:
+            args:
+                model_arg_1: ${HOST:int:8}
+        ...
+        """
+    )
+
+    def teardown(self):
+        """Removes the over-ride"""
+        cwd = self._get_cwd()
+        aea_config_file = Path(cwd, DEFAULT_AEA_CONFIG_FILE)
+        configuration = aea_config_file.read_text()
+        configuration = configuration.replace(self.override, "")
+        aea_config_file.write_text(configuration)
+
+    def _add_skill_override_config(self):
+        """Add custom stub connection config."""
+        cwd = self._get_cwd()
```

Comment:
This is a class-wide directory. So if you manipulate this directory in one test, it will affect the state in another!

Reply:
Yes, I found this out. I also found that most of the tests within this file perform an identical manipulation.
This is what is causing the inconsistent test runs.
I was considering refactoring a number of the other tests in the folder to ensure that the state is reverted post folder manipulation.